kythe/cxx/extractor/cxx_extractor.cc

```diff
@@ -621,14 +654,14 @@ PreprocessorTranscript ExtractorPPCallbacks::PopFile() {
 
 void ExtractorPPCallbacks::EndOfMainFile() {
   if (auto* mfile = GetMainFile()) {
-    AddFile(mfile, std::string(mfile->getName()));
+    *main_source_file_ = AddFile(mfile, mfile->getName());
     *main_source_file_transcript_ = PopFile();
   }
 }
 
 std::string ExtractorPPCallbacks::FixStdinPath(const clang::FileEntry* file,
-                                               const std::string& in_path) {
-  if (in_path == "-" || in_path == "<stdin>") {
+                                               llvm::StringRef path) {
+  if (path == "-" || path == "<stdin>") {
```

Comment:
Any reason why `IsStdinPath` shouldn't be used here?

Reply:
Not really, no.